libass/ass_render.c

```diff
@@ -1479,6 +1479,17 @@ size_t ass_bitmap_construct(void *key, void *value, void *priv)
     return sizeof(BitmapHashKey) + sizeof(Bitmap) + bitmap_size(bm);
 }
 
+static void measure_text_on_eol(
+        const ASS_Renderer *const render_priv, TextInfo *const text_info,
```

Comment:
`render_priv` is unused here, should be removed. Also I don't see any sense in that `const`s everywhere (all except the first). It prohibits variable changes inside a function but doesn't have any effect outside, only complicates a function prototype. And functions long enough to benefit from such a protection should be split to several simpler functions instead.

It's possible to further reduce prototype by eliminating `scale` at the cost of longer invocation, but I'm not sure which variant would be better:
```c
measure_text_on_eol(text_info, cur_line, scale * max_asc, scale * max_desc);
```

Reply:
`render_priv` is later needed for `render_priv->border_scale`.
But `text_info` can be accesed via render_priv, so only one of them is necessary, my bad.

Personally I'd prefer to keep `scale` as an argument, from a style perspective (though I don't mind changing that if requested).  
While pre-multiplying would be a bit more time-efficient (assuming the compiler doesn't optimize it anyway).

Reply:
If it's needed only later then that argument should be added later too. It also can be premultiplied into `max_border_y` to eliminate that dependency altogether, but merging `text_info` and `render_priv` is good too.

Reply:
iinm you'd prefer the following version ?

```C
static void measure_text_set_maxima(TextInfo *text_info, int cur_line,
                                double max_asc_scaled, double max_desc_scaled,
                                double max_bord_y_scaled)
{
    text_info->lines[cur_line].asc  = max_asc_scaled;
    text_info->lines[cur_line].desc = max_desc_scaled;
    text_info->height += max_asc_scaled + max_desc_scaled;
    // For *VSFilter compatibility do biased rounding on max_border*
    text_info->border_bottom = (int) (max_border_y_scaled + 0.5);
    if (cur_line == 0)
        text_info->border_top = text_info->border_bottom;
}
```

Personally, I find the current version cleaner, as the whole logic to scale, cast and update the text_info members correctly is contained in the function. Imho it's also more readable despite the longer function prototype.  *(especially if the rounding and cast of the vertical border was also moved outside the function)*

Ofc your opinion may differ and I'd change it if your not happy with the current version.

Reply:
This version is cleaner conceptually (less context has leaked from outside), your version is simpler (less duplication). So it's about the same for me, I don't have clear preference here.